SlackAPI.Tests/Configuration/IntegrationFixture.cs

```diff
@@ -15,15 +17,39 @@ public class IntegrationFixture : IDisposable
         public IntegrationFixture()
         {
             this.Config = this.GetConfig();
-            this.userClient = new Lazy<SlackSocketClient>(() => this.GetClient(this.Config.UserAuthToken));
-            this.botClient = new Lazy<SlackSocketClient>(() => this.GetClient(this.Config.BotAuthToken));
+            this.userClient = new Lazy<SlackSocketClient>(() => this.CreateClient(this.Config.UserAuthToken));
+            this.botClient = new Lazy<SlackSocketClient>(() => this.CreateClient(this.Config.BotAuthToken));
         }
 
         public SlackConfig Config { get; }
 
-        public SlackSocketClient UserClient => userClient.Value;
+        public SlackSocketClient UserClient
+        {
+            get
+            {
+                Assert.True(userClient.Value.IsReady);
+                return userClient.Value;
+            }
+        }
 
-        public SlackSocketClient BotClient => botClient.Value;
+        public SlackSocketClient BotClient
+        {
+            get
+            {
+                Assert.True(botClient.Value.IsReady);
+                return botClient.Value;
+            }
+        }
+
+        public SlackSocketClient CreateUserClient(IWebProxy proxySettings = null)
```

Comment:
I introduced new methods here to be able to really create new clients and not used cached ones